### DISCUSSION TITLE
How do I use prism in the WPF class library

### DISCUSSION THREAD

**Discussion by zhengyong1023** [PrismLibrary/Prism#2838] [WPF]
I will provide a WPF class library to the customer.  I already have a WPF application with prism.  How do I convert it to a library.   Class libraries cannot use PrismApplication， How can I use a container without PrismApplication?

I hope to get your help. Thank you

**Comment by brianlagunas** [comment]
Designing a class library is different than an application. The application is always responsible for providing the container instance, not the library. You must design your library so that it can be integrated into an application that uses a container.
